client/client.go

```diff
@@ -21,26 +23,37 @@ import (
 	grpc_echo_pb "github.com/datawire/kat-backend/echo"
 	"github.com/gogo/protobuf/proto"
 	"github.com/gorilla/websocket"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
 )
 
-type semaphore chan bool
+// Limit concurrency
 
-func Semaphore(n int) semaphore {
-	sem := make(semaphore, n)
+// Semaphore is a counting semaphore that can be used to limit concurrency.
+type Semaphore chan bool
+
+// NewSemaphore returns a new Semaphore with the specified capacity.
+func NewSemaphore(n int) Semaphore {
+	sem := make(Semaphore, n)
 	for i := 0; i < n; i++ {
 		sem.Release()
 	}
 	return sem
 }
 
-func (s semaphore) Acquire() {
+// Acquire blocks until a slot/token is available.
+func (s Semaphore) Acquire() {
 	<-s
 }
 
-func (s semaphore) Release() {
+// Release returns a slot/token to the pool.
+func (s Semaphore) Release() {
 	s <- true
 }
 
+// rlimit frobnicates the interplexing beacon. Or maybe it reverses the polarity
+// of the neutron flow. I'm not sure. FIXME.
```

Comment:
Is this comment relevant?

Reply:
The `FIXME` at the end is important. The rest is silly. I'd like @rhs or someone else who knows what's going on to fill in the correct comment. Regrettably, the original history for this function is lost, or at least I don't know where to find it.

Reply:
👍